core/tag.h

```diff
@@ -168,56 +168,60 @@ size_t lf_readable_time(char* buffer, instant_t time);
 size_t lf_comma_separated_time(char* buffer, instant_t time);
 
 /**
- * Return the current logical time in nanoseconds.
- * On many platforms, this is the number of nanoseconds
- * since January 1, 1970, but it is actually platform dependent.
- * 
- * @return A time instant.
- */
-DEPRECATED(instant_t get_logical_time(void));
-
-/**
- * Return the current tag, a logical time, microstep pair.
- */
-DEPRECATED(tag_t get_current_tag(void));
-
-/**
- * Return the current microstep.
+ * An enum for specifying the desired tag when calling "lf_time"
  */
-DEPRECATED(microstep_t get_microstep(void));
+typedef enum _lf_time_type {
+    LF_LOGICAL,
+    LF_PHYSICAL,
+    LF_ELAPSED_LOGICAL,
+    LF_ELAPSED_PHYSICAL,
+    LF_START
+} _lf_time_type;
 
 /**
- * Global physical clock offset.
- * Initially set according to the RTI's clock in federated
- * programs.
+ * Get the time specified by "type".
+ * 
+ * Example use cases:
+ * - Getting the starting time:
+ * lf_time_start()
+ * 
+ * - Getting the elapsed physical time:
+ * lf_time_elapsed_physical()
+ * 
+ * - Getting the logical time
+ * lf_time_logical()
+ * 
+ * @param type A field in an enum specifying the time type. 
+ *             See enum "lf_time_type" above.
+ * @return The desired time
  */
-extern interval_t _lf_global_physical_clock_offset;
+instant_t _lf_time(_lf_time_type type);
 
 /**
- * A test offset that is applied to the clock.
- * The clock synchronization algorithm must correct for this offset.
- * This offset is especially useful to test clock synchronization on the
- * same machine.
+ * Return the current logical time in nanoseconds.
+ * On many platforms, this is the number of nanoseconds
+ * since January 1, 1970, but it is actually platform dependent.
+ * 
+ * @return A time instant.
  */
-extern interval_t _lf_global_test_physical_clock_offset;
+instant_t lf_time_logical(void);
+DEPRECATED(instant_t get_logical_time(void));
 
 /**
- * Offset to _LF_CLOCK that would convert it
- * to epoch time. This is applied to the physical clock
- * to get a more meaningful and universal time.
- * 
- * For CLOCK_REALTIME, this offset is always zero.
- * For CLOCK_MONOTONIC, it is the difference between those
- * clocks at the start of the execution.
+ * Return the elapsed logical time in nanoseconds
+ * since the start of execution.
+ * @return A time interval.
  */
-extern interval_t _lf_epoch_offset;
+interval_t lf_time_elapsed_logical(void);
```

Comment:
I recall from this morning that we proposed `lf_time_logical_elapsed` (and `lf_time_physical_elapsed`). I slightly prefer it this ordering as it makes more sense to me when I think about these concepts in a taxonomy. I think it will also work better with autocomplete.

Reply:
Marking this as resolved since it has already been addressed.